hoomd/hpmc/pytest/test_shape_updater.py

```diff
@@ -0,0 +1,264 @@
+        if hasattr(move, attr):
+            assert getattr(move, attr) == value
+
+
+@pytest.mark.parametrize("updater_constructor_args", updater_constructor_args)
+def test_valid_construction_shape_updater(updater_constructor_args):
+
+    updater = hpmc.update.Shape(**updater_constructor_args)
+
+    # validate the params were set properly
+    for attr, value in updater_constructor_args.items():
+        assert getattr(updater, attr) == value
+
+
+@pytest.mark.parametrize("shape_move_obj,attr,value", shape_move_valid_attrs)
+def test_valid_setattr_shape_move(shape_move_obj, attr, value):
+    """Test that the shape move classes can get and set attributes."""
+    setattr(shape_move_obj, attr, value)
+    assert getattr(shape_move_obj, attr) == value
+
+
+@pytest.mark.parametrize("attr,value", shape_updater_valid_attrs)
+def test_valid_setattr_shape_updater(attr, value):
+    """Test that the Shape updater can get and set attributes."""
+    updater = hpmc.update.Shape(trigger=1,
+                                shape_move=ShapeMove(move_probability=1))
+
+    setattr(updater, attr, value)
+    assert getattr(updater, attr) == value
+
+
+@pytest.mark.parametrize("obj,attr,value", type_parameters)
+def test_type_parameters(obj, attr, value):
+    getattr(obj, attr)["A"] = value
+    hoomd.conftest.equality_check(getattr(obj, attr)["A"], value)
+
+
+def test_vertex_shape_move(device, simulation_factory,
+                           two_particle_snapshot_factory):
+
+    verts = np.asarray([[-1, -1, -1], [-1, -1, 1], [-1, 1, -1], [1, -1, -1],
+                        [-1, 1, 1], [1, -1, 1], [1, 1, -1], [1, 1, 1]]) / 2
+
+    move = Vertex()
+    move.volume["A"] = 1
+
+    updater = hpmc.update.Shape(trigger=1,
+                                shape_move=move,
+                                step_size=0.2,
+                                nsweeps=2)
+    updater.shape_move = move
+
+    mc = hoomd.hpmc.integrate.ConvexPolyhedron()
+    mc.d["A"] = 0
+    mc.a["A"] = 0
+    mc.shape["A"] = dict(vertices=verts)
+
+    # create simulation & attach objects
+    sim = simulation_factory(two_particle_snapshot_factory(d=10))
+    sim.operations.integrator = mc
+    sim.operations += updater
+
+    # test attachmet before first run
+    assert not move._attached
+    assert not updater._attached
+    sim.run(0)
+
+    # test attachmet after first run
+    assert move._attached
+    assert updater._attached
+
+    # run with 0 probability of performing a move:
+    #  - shape should remain unchanged
+    #  - all moves accepted
+    move.move_probability = 0
+    sim.run(3)
+    assert np.allclose(mc.shape["A"]["vertices"], verts)
+    assert updater.shape_moves[0] == 6
+    assert updater.shape_moves[1] == 0
+
+    # always attempt a shape move:
+    #  - shape should change, some attemps are accepted but
+    #  - volume should remain unchanged
+    move.move_probability = 1
+    sim.run(10)
+    assert updater.shape_moves[0] != 0
```

Comment:
These kind of checks are finicky as they require that an acceptable move is tried.

Reply:
I agree. Do you suggest removing these kind of checks?

Reply:
Yes, I would prefer that. We can add a validation test to the validation test suite later (we are going to create a separate repo, or at least have tests that only run for releases) that test that we get the expected statistical/thermodynamic results.

Reply:
Sounds good.